app/controllers/TaskController.scala

```diff
@@ -350,6 +353,40 @@ class TaskController @Inject() (implicit val env: Environment[User, SessionAuthe
         env.screenHeight, env.operatingSystem, Some(remoteAddress), env.language, env.cssZoom, Some(currTime))
       AuditTaskEnvironmentTable.save(taskEnv)
 
+      // Adding the new panorama information to the pano_history table.
+      // Sifting through the histories to add the current panorama first 
+      // so foreign key constraint is fulfilled.
+      val panoHistories: Seq[PanoHistorySubmission] = data.panoHistories
+      panoHistories.foreach { panoHistory => 
+        val locationCurrentPanoId: String = panoHistory.currentId
+        val individualHistories: Seq[IndividualPanoHistory] = panoHistory.history
+        val visitedTimestamp: String = panoHistory.visitedTimestamp
+        val loop = new Breaks
+        loop.breakable {
```

Comment:
Instead of making a loop and breaking out of it, let's just get the one that we want and submit it first...
```
val currPanoHistory: IndividualPanoHistory = panoHistory.history(panoHistory.history.indexWhere(_.pano == panoHistory.currentId))
```

Alternatively, you could just cheat and store the current pano's date at the `PanoHistorySubmission` level instead of keeping it in the list of `IndividualPanoHistory` because it's just one or two fields and then you don't have to search for it later :grin: